src/Juvix/Compiler/Core/Transformation/UnrollRecursion.hs

```diff
@@ -1,7 +1,88 @@
 module Juvix.Compiler.Core.Transformation.UnrollRecursion (unrollRecursion) where
 
+import Data.HashMap.Strict qualified as HashMap
+import Juvix.Compiler.Core.Data.IdentDependencyInfo
+import Juvix.Compiler.Core.Data.InfoTableBuilder
+import Juvix.Compiler.Core.Extra
 import Juvix.Compiler.Core.Transformation.Base
 
--- TODO: not yet implemented / at first only check for recursion and give an error
 unrollRecursion :: InfoTable -> InfoTable
-unrollRecursion tab = tab
+unrollRecursion tab =
+  let (mp, (tab', ())) =
+        run $
+          runState @(HashMap Symbol Symbol) mempty $
+            runInfoTableBuilder tab $
```

Comment:
we should add the following to `module Juvix.Compiler.Core.Data.InfoTableBuilder`
```
execInfoTableBuilder :: InfoTable -> Sem (InfoTableBuilder ': r) a -> Sem r InfoTable
execInfoTableBuilder tab = fmap fst . runInfoTableBuilder tab
```
and use it here